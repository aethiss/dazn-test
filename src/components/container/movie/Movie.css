#container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  /*height: 300px;*/
}

.box {
  width: 97%;
  margin: 5px;
  text-align: center;
}

.box img {
  max-width: 60%;
}

.movie-title {
  font-weight: bold;
}

.movie-rating {
  line-height: 1px;
  padding-top: 5px;
}
