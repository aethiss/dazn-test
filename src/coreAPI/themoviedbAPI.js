import {
  BASE_URL,
  APIKEY,
  TRENDING_URL,
  MOVIE_SEARCH,
  MOVIE_URL } from '../constants/movieDbUrls'

export const searchMovie = (getUrl) => {
  return new Promise((resolve, reject) => {
    fetch(getUrl, {
      method: 'GET',
      headers: {
        'Accept': 'application/json, text/plain, */*',
        'Content-Type': 'application/json'
      },
    })
    .then((resp) => resp.json())
      .then((data) => {
        // console.log('DATA ===> ', data)
        resolve(data)
      })
      .catch((err) => {
        // console.log('error ::::: ', err)
        reject(err)
      })
  })
}

export const generateRequestUrl = (type = 'trending', value = '') => {
  switch(type) {
    case 'search':
      return `${BASE_URL}${MOVIE_SEARCH}?api_key=${APIKEY}&query=${value}`
      break;
    case 'details':
      // code block movie_id
      return `${BASE_URL}${MOVIE_URL}/${value}?api_key=${APIKEY}`
      break;
    default:
    return `${BASE_URL}${TRENDING_URL}?api_key=${APIKEY}`
  }
}
