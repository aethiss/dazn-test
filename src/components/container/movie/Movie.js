import React, { Component } from 'react'
import {generateRequestUrl, searchMovie} from "../../../coreAPI/themoviedbAPI"
import { BASE_IMG_URL_500 } from "../../../constants/movieDbUrls"

import Style from './Movie.css'

export default class Movie extends Component {
  constructor(props) {
    super(props)
    this.state = {
      movie: false
    }
  }

  componentDidMount() {
    const { match } = this.props
    const movieId = match.params ? match.params.movieId : false

    searchMovie(generateRequestUrl('details', movieId)).then((res) => {
      this.setState({
        movie: res
      })
    })
  }

  render() {
    const { movie } = this.state
    return (
      <div>
        { !this.state.movie ?
          <div>...Loading</div> :
          <div className="movie-containe">
            <div className="box movie-title">
              <p>{movie.original_title}</p>
            </div>
            <div className="box">
              <img src={`${BASE_IMG_URL_500}${movie.poster_path}`} />
            </div>
            <div className="box movie-description">
              <p>{movie.overview}</p>
            </div>
            <div className="box movie-rating">
              {
                movie.genres && movie.genres.map((genre, key) => {
                  return <p key={key}>{genre.name}</p>
                })
              }
            </div>
          </div>
        }
      </div>
    )
  }

}
