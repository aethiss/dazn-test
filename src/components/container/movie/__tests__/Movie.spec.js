import React from 'react'
import { shallow, mount } from 'enzyme'

import Movie from '../Movie'
import * as api from '../../../../coreAPI/themoviedbAPI'
const mySpy = jest.spyOn(api, 'searchMovie').mockImplementation(() => {
  return Promise.resolve({ results: [] })
})

const mockProps = {
  match: {
    params: {
      movieId: 12345
    }
  }
}

describe('@Home', () => {
  describe('render', () => {
    it('should render default home page', () => {
      const component = shallow(<Movie {...mockProps} />)
      expect(component).toMatchSnapshot()
    })
  })
  describe('lifecycle', () => {
    it('should call searchMovie when the component is mounted', () => {
      mount(<Movie {...mockProps} />)
      expect(mySpy).toHaveBeenCalledWith('http://api.themoviedb.org/3/movie/12345?api_key=f70db02dd395f7b2f0e354374b13d4ca')
    })
  })
})
