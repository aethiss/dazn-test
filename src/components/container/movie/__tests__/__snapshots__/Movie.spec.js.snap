// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Home render should render default home page 1`] = `
<div>
  <div>
    ...Loading
  </div>
</div>
`;
